optaplanner-core/src/main/java/org/optaplanner/core/config/solver/SolverConfig.java

```diff
@@ -257,6 +293,37 @@ protected RandomFactory buildRandomFactory(EnvironmentMode environmentMode_) {
         return randomFactory;
     }
 
+    protected Integer resolveMoveThreadCount() {
+        int availableProcessorCount = Runtime.getRuntime().availableProcessors();
+        Integer resolvedMoveThreadCount;
+        if (moveThreadCount == null || moveThreadCount.equals(MOVE_THREAD_COUNT_NONE)) {
+            return null;
+        } else if (moveThreadCount.equals(MOVE_THREAD_COUNT_AUTO)) {
+            // Leave one for the Operating System and 1 for the solver thread, take the rest
+            resolvedMoveThreadCount = (availableProcessorCount - 2);
+            if (resolvedMoveThreadCount <= 1) {
```

Comment:
1 move thread != single threaded. In single-threaded there are no move threads. With 1 move thread, there are 2 threads, because there is also the solver thread.

For AUTO, we replace 1 move thread with single threaded. Comment improved to make this clearer:
// To deliberately enforce 1 moveThread, set the moveThreadCount explicitly to 1.